src/Eshopworld.WorkerProcess/Configuration/Autofac/ContainerBuilderExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Autofac;
+using Eshopworld.Core;
+using EShopworld.WorkerProcess.CosmosDistributedLock;
+using EShopworld.WorkerProcess.Stores;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace EShopworld.WorkerProcess.Configuration.Autofac
+{
+    public static class ContainerBuilderExtensions
+    {
+        private const string CosmosDataStoreOptions = "WorkerProcess:CosmosDataStore";
+        /// <summary>
+        /// Adds <see cref="DistributedLock"/> to the services collection
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="configuration"></param>
+        public static void AddCosmosDistributedLock(this ContainerBuilder builder, IConfigurationRoot configuration)
+        {
+            var cosmosDataStoreOptions = configuration.BindSection<CosmosDataStoreOptions>(CosmosDataStoreOptions,
+                m => { m.AddMapping(x => x.ConnectionString, "cm--cosmos-connection--worker-process"); });
+
+            builder.Register(ctx => Options.Create(cosmosDataStoreOptions))
```

Comment:
We are registering this just to resolve it later in line 33, can we not just inject it there?

Reply:
We can certainly use it in line 33, but registration is required by DistributedLock

Reply:
I  am not getting you.
I mean to remove line 24 and change line 30 to look like this:
```
builder.Register(ctx =>
    new CosmosDistributedLockStore(
        documentClient,
        Options.Create(cosmosDataStoreOptions),
        ctx.Resolve<IBigBrother>())).As<ICosmosDistributedLockStore>().SingleInstance();

```

Reply:
ok but how will IOptions<CosmosDataStoreOptions> be resolved in DistributedLock constructor? If we remove registration?

Reply:
Ah yes sorry I missed that on the last line!